### PR TITLE
Some more examples/refs for array docs

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: base/abstractarray.jl [hunk 1]
```diff
@@ -968,6 +968,22 @@ _unsafe_ind2sub(sz, i) = (@_inline_meta; _ind2sub(sz, i))
     setindex!(A, X, inds...)
 
 Store values from array `X` within some subset of `A` as specified by `inds`.
```
Comment (fredrikekre) [inline base/abstractarray.jl]: While you are at it; append something like
```
The syntax `A[inds...] = X` (lowers|is equivalent) to `setindex!(A, X, inds...)`
```
?
Perhaps also change the signature up top to
```
    setindex!(A, X, inds...)
    A[inds...] = X

Store values ...
```

File: base/abstractarray.jl [hunk 2]
```diff
@@ -968,6 +968,22 @@ _unsafe_ind2sub(sz, i) = (@_inline_meta; _ind2sub(sz, i))
...
+
+# Examples
+```jldoctest
+julia> A = zeros(2, 2, 2, 2);
```
Comment (fredrikekre) [inline base/abstractarray.jl]: Use a smaller matrix such that we can show if all after `setindex!`ing? Perhaps also unnecessary to use `CartesianIndex`? Proposal:
````
```jldoctest
julia> A = zeros(2,2);

julia> setindex!(A, [10, 20], [1, 2]);

julia> A[[3, 4]] = [30, 40];

julia> A
2×2 Array{Float64,2}:
 10.0  30.0
 20.0  40.0
```
````

File: base/array.jl [hunk 3]
```diff
@@ -127,15 +136,45 @@ asize_from(a::Array, n) = n > ndims(a) ? () : (arraysize(a,n), asize_from(a, n+1
...
 
 """
     Base.bitsunionsize(U::Union)
 
-For a Union of `isbitstype` types, return the size of the largest type; assumes `Base.isbitsunion(U) == true`
+For a `Union` of [`isbitstype`](@ref) types, return the size of the largest type; assumes `Base.isbitsunion(U) == true`.
+
+# Examples
+```jldoctest
+julia> ut = Union{Float64, UInt8}
```
Comment (fredrikekre) [inline base/array.jl]: `;`

File: base/array.jl [hunk 4]
```diff
@@ -127,15 +136,45 @@ asize_from(a::Array, n) = n > ndims(a) ? () : (arraysize(a,n), asize_from(a, n+1
...
+Union{Float64, UInt8}
+
+julia> Base.bitsunionsize(ut)
+0x0000000000000008
+
+julia> ut = Union{Float64, UInt8, Int128}
```
Comment (fredrikekre) [inline base/array.jl]: `;`

File: base/array.jl [hunk 5]
```diff
@@ -127,15 +136,45 @@ asize_from(a::Array, n) = n > ndims(a) ? () : (arraysize(a,n), asize_from(a, n+1
...
+
+# Examples
+```jldoctest
+julia> ut = Union{Float64, UInt8}
+Union{Float64, UInt8}
+
+julia> Base.isbitsunion(ut)
+true
+
+julia> ut = Union{Float64, String}
```
Comment (fredrikekre) [inline base/array.jl]: `;`

File: base/array.jl [hunk 6]
```diff
@@ -127,15 +136,45 @@ asize_from(a::Array, n) = n > ndims(a) ? () : (arraysize(a,n), asize_from(a, n+1
 """
     Base.isbitsunion(::Type{T})
 
-Return whether a type is an "is-bits" Union type, meaning each type included in a Union is `isbitstype`.
+Return whether a type is an "is-bits" Union type, meaning each type included in a Union is [`isbitstype`](@ref).
+
+# Examples
+```jldoctest
+julia> ut = Union{Float64, UInt8}
```
Comment (fredrikekre) [inline base/array.jl]: `;`

Comment (KristofferC) [reply base/array.jl]: Also, is there a need to create the variable at all here?

Comment (fredrikekre) [reply base/array.jl]: No, probably cleaner and more concise without it.

**Pull request merged**